nodewatcher/nodewatcher.py

```diff
@@ -131,22 +133,24 @@ def _lock_host(scheduler_module, hostname, unlock=False):
     time.sleep(15)  # allow for some settling
 
 
-def _self_terminate(asg_name, asg_client, instance_id):
+def _self_terminate(asg_client, instance_id, decrement_desired=True):
     """
     Terminate the given instance and decrease ASG desired capacity.
 
-    :param asg_name: ASG name
     :param asg_client: ASG boto3 client
-    :param instance_id: the instnace to terminate
+    :param instance_id: the instance to terminate
+    :param decrement_desired: if True decrements ASG desired by 1
     """
-    if _maintain_size(asg_name, asg_client):
-        log.info("Not terminating due to min cluster size reached")
+    try:
+        log.info("Self terminating %s" % instance_id)
+        asg_client.terminate_instance_in_auto_scaling_group(
+            InstanceId=instance_id, ShouldDecrementDesiredCapacity=decrement_desired
+        )
+        sys.exit(0)
```

Comment:
The `sys.exit(0)` is not really useful since the supervisord will restart the daemon, but ok we can leave it.

Reply:
supervisord by default does not restart the program when exitcode is 0 and we do not override this config value

source: http://supervisord.org/configuration.html - exitcodes (defaults to 0)